appcues/src/main/java/com/appcues/action/appcues/SubmitFormAction.kt

```diff
@@ -0,0 +1,63 @@
+package com.appcues.action.appcues
+
+import com.appcues.Appcues
+import com.appcues.action.ExperienceAction
+import com.appcues.action.ExperienceActionQueueTransforming
+import com.appcues.analytics.AnalyticsTracker
+import com.appcues.analytics.ExperienceLifecycleEvent.StepInteraction
+import com.appcues.analytics.ExperienceLifecycleEvent.StepInteraction.InteractionType.FORM_SUBMITTED
+import com.appcues.analytics.formattedAsProfileUpdate
+import com.appcues.data.model.AppcuesConfigMap
+import com.appcues.data.model.getConfigOrDefault
+import com.appcues.statemachine.StateMachine
+
+internal class SubmitFormAction(
+    override val config: AppcuesConfigMap,
+    private val stateMachine: StateMachine,
+    private val analyticsTracker: AnalyticsTracker,
+) : ExperienceActionQueueTransforming {
+
+    companion object {
+        const val TYPE = "@appcues/submit-form"
+    }
+
+    private val skipValidation = config.getConfigOrDefault("skipValidation", false)
+
+    // validate form and block future actions if needed
+    override fun transformQueue(queue: List<ExperienceAction>, index: Int, appcues: Appcues): List<ExperienceAction> {
```

Comment:
checking to see if I understood it.

if the form has validations and if isFormComplete is false we are dropping all other actions as a way to preventing actions from running when form is not yet submitted?